02-clients/java/03-java-tutorial.md

```diff
@@ -0,0 +1,161 @@
+---
+pageTitle: Java Driver tutorial
+keywords: typedb, client, java
+longTailKeywords: typedb java client, typedb client java, client java, java client
+Summary: Tutorial for TypeDB Java Driver.
+---
+
+# Java Driver tutorial
+
+## Prerequisites
+
+Make sure to install the Java Driver as per [Installation](02-java-install.md) page. The TypeDB server is also 
+needs to be installed and running.
+
+## Importing Java Driver and connecting to a server
+
+Import the following modules: 
+
+- `com.vaticle.typedb.client.TypeDB`
+- `com.vaticle.typedb.client.api.TypeDBClient`
+- `com.vaticle.typedb.client.api.TypeDBSession`
+
+## Connecting
+
+Instantiate a TypeDB Core client and open a [session](../../0001-typedb/02-dev/01-connect.md#sessions) to a 
+[database](../../0001-typedb/02-dev/01-connect.md#databases). 
+
+<div class="note">
+[Note]
+If you don't have a database, you can find how to create one on the 
+[Quickstart](../0001-typedb/01-start/03-quickstart.md#create-a-database) page.
+
+The name for the database should be `social_network`.
+</div>
+
+<!-- test-example TypeDBQuickstartA.java -->
+```java
+package com.vaticle.doc.examples;
+
+import com.vaticle.typedb.client.TypeDB;
+import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBSession;
+
+public class TypeDBQuickstartA {
+    public static void main(String[] args) {
+        TypeDBClient client = TypeDB.coreClient("localhost:1729");
+        // client is open
+        TypeDBSession session = client.session("social_network", TypeDBSession.Type.DATA);
+        // session is open
+        session.close();
+        // session is closed
+        client.close();
+        // client is closed
+    }
+}
+```
+
+## Creating transactions
+
+Create transactions to use for reading and writing data.
+
+<!-- test-example TypeDBQuickstartB.java -->
+```java
+package com.vaticle.doc.examples;
+
+import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBSession;
+import com.vaticle.typedb.client.api.TypeDBTransaction;
+import com.vaticle.typedb.client.TypeDB;
+
+public class TypeDBQuickstartB {
+    public static void main(String[] args) {
+        TypeDBClient client = TypeDB.coreClient("localhost:1729");
+
+        try (TypeDBSession session = client.session("social_network", TypeDBSession.Type.DATA)) {
+            // creating a write transaction
+            TypeDBTransaction writeTransaction = session.transaction(TypeDBTransaction.Type.WRITE);
+            // write transaction is open
+            // write transaction must always be committed (closed)
+            writeTransaction.commit();
+    
+            // creating a read transaction
+            TypeDBTransaction readTransaction = session.transaction(TypeDBTransaction.Type.READ);
+            // read transaction is open
+            // read transaction must always be closed
+            readTransaction.close();
+        }
```

Comment:
Why not use the `try (...)` construct for transactions as well?

Reply:
We doing just that down there. I guess the logic was to keep the transaction as a one-liner at this stage. To simplify the code. I think it can help someone to see both variants (with and without the try() block).